solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -84,11 +88,12 @@ contract InterchainAccountRouter is Router, IInterchainAccountRouter {
         return bytes32(abi.encodePacked(_origin, _sender));
     }
 
-    function _getInterchainAccount(bytes32 salt)
+    function _getInterchainAccount(bytes32 salt, bytes memory bytecode)
         internal
         view
         returns (address)
     {
+        bytes32 bytecodeHash = keccak256(bytecode);
```

Comment:
Should we write this to storage? The implementation address/bytecode can't change, can it?